### DISCUSSION TITLE
how to set zindex of the control rectangle so that it stays on top on any element on the page or canvas?

### DISCUSSION THREAD

**Discussion by geminigeek** [fabricjs/fabric.js#9367] [Q&A]
hi,

my question is same as the title how to set zindex of the control rectangle so that they stay on top on any element on the page or canvas?

i tried many approach to do this , but its not working, i tried controlsAboveOverlay this works for the other objects of the fabric, not my use case.

my use case is i have a transparent rect and on top of it i have a custom html div like this example http://fabricjs.com/interaction-with-objects-outside-canvas have a button, 

the div is same size as rect it covers rect completely at all times,  if i place two of these on top of each other one big (back in my example) and another small (pink)  i can only select the back/big one, the small one's controls are hidden behind the big one. i am pasting a pic to show how the controls are behind

![image](https://github.com/fabricjs/fabric.js/assets/3751322/1ecde433-254f-4c7a-98f9-8754ff001e3c)

is there a way where the control of pink object can be made on top of the black object ?

**Comment by ShaMan123** [comment]
You can't expect something rendered on a canvas to render above a html elements that is on top of the canvas
It is like having a fish in a bowl, a wall between you and the bowl and you looking at the wall and hoping to see the fish and for it to swim in front if you

**Reply by geminigeek** [reply]
Ha ha ha, yea even I thought so, I am learning frontend so it's new to me, I already have a solution to pass drag event on the element to trigger the rect position behind it, which will fire moving to the front element again ,but it felt like over engineering , anyways thanks for your reply

**Comment by ShaMan123** [comment]
You can however place another fish bowl in front of the wall and trick yourself to think that the fish you see is the fish from behind the wall.
Try inserting your div in between the 2 canavses fabric uses and rebder the controls on the top cnavas

**Reply by asturur** [reply]
render the controls on the top canvas is an old goal anyway that at some point i will get to.
Re-rendering the whole canvas to add controls is an overkill anyway
